src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENT_DATA = {};
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = EMPTY_ELEMENT_DATA;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheckForAdd(index);
+        int countElementsAfterIndex = size - index;
+        Object[] elementData;
+        if (size == (elementData = this.elementData).length) {
+            elementData = grow();
+        }
+        System.arraycopy(elementData, index,
+                elementData, index + 1,
+                countElementsAfterIndex);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int amountListElements = list.size();
+        Object [] arrayFromList = toArray(list, amountListElements);
```

Comment:
```suggestion
        Object [] arrayFromList = toArray(list);
```
you can get the size inside toArray method